assignments/array-methods.js

```diff
@@ -55,29 +55,61 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 
 // ==== Challenge 1: Use .forEach() ====
 // The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
-let fullName = [];
-console.log(fullName);
+// let fullName = [];
+
+// runners.forEach(function(element){
+//     let name = `${element.first_name} ${element.last_name}`
+//     fullName.push(name);
+// });
+
+// console.log(fullName);
+// console.log(runners);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
-let allCaps = [];
-console.log(allCaps); 
+// const allCaps = runners.map(function(element){
+//     return element.first_name.toUpperCase()
+// });
+// console.log(allCaps); 
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
-let largeShirts = [];
-console.log(largeShirts);
+// const largeShirts = runners.filter(function(element){
+//     return element.shirt_size === "L";
+// });
+
+// console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
-let ticketPriceTotal = [];
-console.log(ticketPriceTotal);
+// const ticketPriceTotal = runners.reduce(function(totalValue, currentValue){
+//     return totalValue + currentValue.donation;
+// }, 0);
+
+// console.log(ticketPriceTotal);
 
 // ==== Challenge 5: Be Creative ====
 // Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to create and then solve 3 unique problems using one or many of the array methods listed above.
 
-// Problem 1
+// Problem 1 -- use .map() to change dollars to euros 
+// const euroDonations = runners.map(function(element) {
+//     return element.donation * 0.89;
+// });
+
+// console.log(euroDonations);
+
+// Problem 2 -- use .filter() to find out who donated more than 100 dollars
+// const largeDonation = runners.filter(function(element) {
+//     return element.donation > 100;
+// });
+
+// console.log(largeDonation);
+
+// Problem 3 -- use .forEach() to create mailing list 
 
-// Problem 2
+// let mailList = []
+// runners.forEach(function(element){
+//     mailList.push(element.email);
+// })
```

Comment:
These are excellent problems. Very real world! Your code is clean and concise too!